src/utils/risk.ts

```diff
@@ -0,0 +1,275 @@
+
+    const valueList = monthKeys.map(
+        (key) => (
+            selectedMonths[key]
+                ? riskDataItem?.[monthNumberToNameMap[key]]
+                : undefined
+        ),
+    );
+
+    if (aggregationMode === 'max') {
+        return maxSafe(valueList);
+    }
+
+    return sumSafe(valueList);
+}
+
+type GetCountrySeasonal = paths['/api/v1/country-seasonal/']['get'];
+type CountrySeasonal = GetCountrySeasonal['responses']['200']['content']['application/json'];
+type IpcData = CountrySeasonal[number]['ipc_displacement_data'];
+type GwisData = CountrySeasonal[number]['gwis'];
+type IpcEstimationType = components['schemas']['EstimationTypeEnum'];
+
+const estimationPriorityMap: Record<IpcEstimationType, number> = {
+    current: 0,
+    first_projection: 1,
+    second_projection: 2,
+};
+
+export function getPrioritizedIpcData(data: IpcData) {
+    const sortedData = data?.map(
+        (item) => {
+            if (isFalsyString(item.estimation_type) || item.estimation_type === '') {
+                return undefined;
+            }
+
+            if (isFalsyString(item.analysis_date) || isNotDefined(item.total_displacement)) {
+                return undefined;
+            }
+
+            return {
+                ...item,
+                estimation_type: item.estimation_type,
+                analysis_date: item.analysis_date,
+                estimation_priority: estimationPriorityMap[item.estimation_type],
+                analysisTimestamp: new Date(item.analysis_date).getTime(),
+                total_displacement: item.total_displacement,
+            };
+        },
+    ).filter(isDefined).sort((a, b) => (
+        compareNumber(a.year, b.year)
+        || compareNumber(a.month, b.month)
+        || compareNumber(a.estimation_priority, b.estimation_priority)
+        || compareNumber(a.analysisTimestamp, b.analysisTimestamp, -1)
+    )) ?? [];
+
+    const uniqueData = unique(
+        sortedData,
+        (ipcDataItem) => `${ipcDataItem.year}-${ipcDataItem.month}`,
+    );
+
+    return uniqueData;
+}
+
+export function getAverageIpcData(uniqueData: IpcData) {
+    const monthGroupedIpcData = listToGroupList(
+        uniqueData,
+        (datum) => datum.month,
+        (datum) => datum.total_displacement,
+    );
+
+    const ipcRiskDataItem = mapToMap(
+        monthGroupedIpcData,
+        (key) => monthNumberToNameMap[Number(key) - 1],
+        (item) => avgSafe(item),
+    );
+
+    const monthlyValueList = Object.values(ipcRiskDataItem ?? {});
+    const annual_average = avgSafe(monthlyValueList);
+
+    return {
+        ...ipcRiskDataItem,
+        annual_average,
+    };
+}
+
+export function getFiRiskDataItem(data: IpcData | undefined) {
+    if (isNotDefined(data) || data.length === 0) {
+        return undefined;
+    }
+
+    const uniqueData = getPrioritizedIpcData(data);
+    const ipcRiskDataItem = getAverageIpcData(uniqueData);
+    const firstData = data[0];
```

Comment:
Why are we using the first data?

Reply:
To get the common elements of the items. We use any of the item from 0 to length of the list. 0 is just arbitrary here.

Reply:
Let's add NOTE